waku/waku_relay/protocol.nim

```diff
@@ -210,8 +204,25 @@ proc subscribe*(w: WakuRelay, pubsubTopic: PubsubTopic, handler: WakuRelayHandle
     else:
       return handler(pubsubTopic, decMsg.get())
 
+  # rejects messages that are not WakuMessage
+  let wrappedValidator = proc(pubsubTopic: string, 
+                              message: messages.Message): Future[ValidationResult] {.async.} =
+    # can be optimized by checking if the message is a WakuMessage without allocating memory
+    # see nim-libp2p protobuf library
+    let msgRes = WakuMessage.decode(message.data)
+    if msgRes.isErr():
+      return ValidationResult.Reject
+    let msg = msgRes.get()
+
+    # now sequentially validate the message
+    if w.wakuValidators.hasKey(pubsubTopic):
+      for validator in w.wakuValidators[pubsubTopic]:
+        if (await validator(pubsubTopic, msg)) == ValidationResult.Reject:
+          return ValidationResult.Reject
```

Comment:
almost forgot, guess we should handle `Ignore`?

https://github.com/status-im/nim-libp2p/blob/b76bac752ff0305730472a334bd2aaba1bd29210/libp2p/protocols/pubsub/errors.nim#L8

so if a validator returns `Ignore`, also return that result

Reply:
```
let result = await validator(pubsubTopic, msg)
if result != ValidationResult.Accept:
   return result
```

or something like this. `Reject` or `Ignore` returns early. `Acept` forces to jump in the next iteration